### PR TITLE
Optimize pair

### PR DESCRIPTION
**Pull request by catalinnnn**

### CODE REVIEW

File: common/modules/token_send/src/token_send.rs [hunk 1]
```diff
@@ -66,38 +86,36 @@ pub trait TokenSendModule {
...
-        };
         let arg_buffer = ManagedArgBuffer::new_empty(self.type_manager());
-        let endpoint_name = ManagedBuffer::new_from_bytes(self.type_manager(), function);
         let mut payments = ManagedVec::new();
         payments.push(EsdtTokenPayment::new(token.clone(), nonce, amount.clone()));
 
+        let gas_limit: u64;
+        let function: ManagedBuffer;
+        let accept_funds_func = opt_accept_funds_func.clone().into_option();
+        if accept_funds_func.is_some() {
```
Comment (claudiulataretu) [inline common/modules/token_send/src/token_send.rs:102]: I think it is cleaner to have this implementation
`
        let (function, gas_limit) = match opt_accept_funds_func {
            OptionalArg::Some(accept_funds_func) => (
                accept_funds_func.as_managed_ref(),
                self.transfer_exec_gas_limit().get(),
            ),
            OptionalArg::None => (ManagedBuffer::new().as_managed_ref(), 0u64),
        };
`

Comment (catalinnnn) [reply common/modules/token_send/src/token_send.rs:102]: that was my initial intention. couldn't figure out the syntax. you can commit this if you want. it is optional ofc

File: locked-asset/factory/src/locked_asset.rs [hunk 2]
```diff
@@ -132,7 +132,7 @@ pub trait LockedAssetModule: token_supply::TokenSupplyModule + token_send::Token
         let mut percents_sum: u8 = 0;
         let mut last_milestone_unlock_epoch: u64 = 0;
 
-        for milestone in unlock_milestones.into_iter() {
+        for milestone in unlock_milestones.to_vec().into_iter() {
```
Comment (claudiulataretu) [inline locked-asset/factory/src/locked_asset.rs]: is the above `unlock_milestones.into_iter()` the same as this one? (`unlock_milestones.to_vec().into_iter()`)

**Review by claudiulataretu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

File: common/modules/token_send/src/token_send.rs (hunk 1 above)
Comment (sasurobert) [reply common/modules/token_send/src/token_send.rs:74]: +1

File: wat-build.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply wat-build.sh:1]: it's what we use to check up on wee_alloc

File: locked-asset/factory/src/locked_asset.rs (hunk 2 above)
Comment (andrei-marinica) [reply locked-asset/factory/src/locked_asset.rs]: Yes

File: common/modules/token_send/src/token_send.rs [hunk 4]
```diff
@@ -8,55 +8,47 @@ pub trait TokenSendModule {
...
-                let existing_index = compact_payments.iter().position(|x| {
-                    x.token_identifier == payment.token_identifier
-                        && x.token_nonce == payment.token_nonce
-                });
-
-                match existing_index {
-                    Some(index) => compact_payments[index].amount += &payment.amount,
-                    None => compact_payments.push(payment.clone()),
-                }
+            if payment.amount > 0u32 {
```
Comment (yzhang90) [inline common/modules/token_send/src/token_send.rs:44]: why not merge the token with the same token_id in the new implementation?

**Pull request merged**